python3Packages.pytensor: fix test_tri_nonconcrete failure with JAX 0.8.2

kirillrdy opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [ ] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Aleksanaa commented on pkgs/development/python-modules/pytensor/default.nix:
```diff
@@ -80,10 +80,15 @@ buildPythonPackage rec {
 
-  disabledTests = lib.optionals stdenv.hostPlatform.isDarwin [
+  disabledTests = [
+    # TypeError: Shapes must be 1D sequences of concrete values of integer type
```
How do we ensure this doesn't reflect our packaging issues, such as providing an inappropriate dependency? Is there an issue upstream?

kirillrdy replied:
yeah its possible, i'll have a closer look,  there are over 2000+ tests that do pass, so even if it's dependency issue, should be small impact

kirillrdy replied:
test was failing because new version of jax change type of exception that was being raised, created PR upstream

kirillrdy committed:
python3Packages.pytensor: fix test_tri_nonconcrete failure with JAX 0.8.2

The test `test_tri_nonconcrete` was failing because JAX 0.8.2 raises `TypeError` (ConcretizationTypeError) immediately instead of an `AttributeError` as expected by the test for older JAX versions.

This patch allows both exceptions, making the test robust to JAX versions.

Upstream PR: https://github.com/pymc-devs/pytensor/pull/1805

GaetanLepage commented on pkgs/development/python-modules/pytensor/default.nix:
```diff
@@ -80,6 +90,7 @@ buildPythonPackage rec {
 
   # Ensure that the installed package is used instead of the source files from the current workdir
   preCheck = ''
+    mkdir -p $HOME/.pytensor
```
Thanks! Just did.
